### DISCUSSION TITLE
Is it possible to find date related data for updated key-values?

### DISCUSSION THREAD

**Discussion by YashAnand1** [etcd-io/etcd#16254] [Q&A] [answered]
Hello, everyone.

I would love to know if etcd stores information related to *when* a key was updated with a new value. Is there any etcdctl command that can be used to find such information?

I came across the following command for finding the date related information for an updated key-value pair:
>` etcdctl get <key> --write-out=json | jq -r '.kvs[0].mod_revision | tonumber | strftime("%Y-%m-%d %H:%M:%S")' `

However, the output of running the above command for the "name" key was incorrect and as follows:
>` 1970-01-01 00:33:03 `

Clearly, the above method does not work but it does help understand the kind of output I am expecting: **date & time of when a key was updated with a value**.

Please let me know how it will be possible to get such data. Any help would be highly appreciated so thank you in advance.

**Comment by jmhbnz** [comment, marked as answer]
Hey @yash-anand-fosteringlinux - Short answer is that etcd stores revisions as integers, not timestamps on the creation of a key, so this is not really possible from a time perspective.

Also refer https://stackoverflow.com/questions/76692414/is-it-possible-to-retrieve-creation-date-time-of-a-key-value-pair-in-etcd 

If you are using etcd in the context of kubernetes consider exploring kubernetes api server audit logs for timestamp based information.
